controller/volume_controller.go

```diff
@@ -543,6 +543,15 @@ func (vc *VolumeController) ReconcileEngineReplicaState(v *longhorn.Volume, e *l
 			vc.eventRecorder.Eventf(v, v1.EventTypeNormal, EventReasonHealthy, "volume %v became healthy", v.Name)
 		}
 
+		// If disabled replica rebuild, skip eviction and data locality.
+		disableReplicaRebuild, err := vc.ds.GetSettingAsBool(types.SettingNameDisableReplicaRebuild)
```

Comment:
Can we add the logic in `replenishReplicas` instead? We can check if the `rs` is empty to decide if it's the first time creation.

Reply:
Will fix it. Thanks.